src/Frontend/src/composables/formatter.ts

```diff
@@ -76,3 +76,13 @@ function formatTimeValue(timeValue: number, displayTwoDigits = false) {
   const strValue = Math.floor(timeValue);
   return `${displayTwoDigits ? ("0" + strValue).slice(-2) : strValue.toLocaleString()}`;
 }
+
+export function timeSpanToDuration(timeSpan: string | undefined) {
+  if (!timeSpan) return dayjs.duration("PT0S");
+
+  const [days, rest] = timeSpan.split(".");
+  const [hours, minutes, seconds, milliseconds] = rest.split(/[:.]/);
+  const units = { days: parseInt(days) ?? 0, hours: parseInt(hours) ?? 0, minutes: parseInt(minutes) ?? 0, seconds: parseInt(seconds) ?? 0, milliseconds: parseInt(milliseconds) ?? 0 };
```

Comment:
It seems we are also doing this parsing in https://github.com/Particular/ServicePulse/blob/master/src/Frontend/src/composables/deliveryDelayParser.ts
We should reuse the existing function with a bit of refactoring/renaming 😉